src/Compilers/CSharp/Portable/Symbols/TypeParameterSymbol.cs

```diff
@@ -451,8 +451,16 @@ internal static bool IsReferenceTypeFromConstraintTypes(ImmutableArray<TypeWithA
             return result;
         }
 
-        internal static bool? IsNotNullableIfReferenceTypeFromConstraintType(TypeWithAnnotations constraintType)
+        internal static bool? IsNotNullableIfReferenceTypeFromConstraintType(TypeWithAnnotations constraintType, out bool isNonNullableValueType)
```

Comment:
>bool? [](start = 24, length = 5)

I don't know if it's worth the effort to refactor, but it feels like we're trying to return a `Nullable<NullableAnnotation>` (3 possible annotations when applicable + 1 case for not applicable)

Reply:
> I don't know if it's worth the effort to refactor, but it feels like we're trying to return a Nullable<NullableAnnotation> (3 possible annotations when applicable + 1 case for not applicable)

The returned value is applicable in all cases. It is convenient to return an extra information that we already calculated here and avoid recalculating it again.

---
In reply to: [292128942](https://github.com/dotnet/roslyn/pull/36269#discussion_r292128942) [](ancestors = 292128942)